File: BOJ/1018. 체스판 다시 칠하기/체스판 다시 칠하기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    private static int N, M;
    private static int minCount = Integer.MAX_VALUE;
    private static char[][] board;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        N = Integer.parseInt(st.nextToken());
        M = Integer.parseInt(st.nextToken());
        board = new char[N][M];
        for (int i = 0; i < N; i++) {
            String line = br.readLine();
            for (int j = 0; j < M; j++) {
                board[i][j] = line.charAt(j);
            }
        }

        for (int i = 0; i <= N - 8; i++) {
            for (int j = 0; j <= M - 8; j++) {
                makeChessBoard(i, j);
            }
        }
        System.out.println(minCount);
    }

    private static void makeChessBoard(int startX, int startY) {
        char[] case1 = new char[] {'B', 'W', 'B', 'W', 'B', 'W', 'B', 'W'};
        char[] case2 = new char[] {'W', 'B', 'W', 'B', 'W', 'B', 'W', 'B'};

        countChessRepaints(startX, startY, case1, case2);
        countChessRepaints(startX, startY, case2, case1);
    }

    private static void countChessRepaints(int startX, int startY, char[] case1, char[] case2) {
        int count = 0;

        for (int i = startX; i < startX + 8; i++) {
            for (int j = startY; j < startY + 8; j++) {
                if (i % 2 == 0) {
                    if (board[i][j] != case1[j - startY]) count += 1;
                } else {
                    if (board[i][j] != case2[j - startY]) count+= 1;
                }
            }
        }
        minCount = Math.min(count, minCount);
    }
}

File: BOJ/1018. 체스판 다시 칠하기/체스판 다시 칠하기.py
n, m = map(int, input().split())

board = [input() for _ in range(n)]

def paint(board, i, j, color):
    count = 0
    for l in range(i, i + 8):
        for m in range(j, j + 8):
            if l % 2 == 0:
                if m % 2 == 0:
                    if board[l][m] != color:
                        count += 1
                else:
                    if board[l][m] == color:
                        count += 1
            else:
                if m % 2 == 0:
                    if board[l][m] == color:
                        count += 1
                else:
                    if board[l][m] != color:
                        count += 1
    return count

min_count = float('inf')
for i in range(n - 7):
    for j in range(m - 7):
        min_count = min(min_count, paint(board, i, j, 'B'), paint(board, i, j, 'W'))

print(min_count)